packages/next/server/next-server.ts

```diff
@@ -1193,3 +1202,24 @@ export default class NextNodeServer extends BaseServer {
     return require(join(this.distDir, ROUTES_MANIFEST))
   }
 }
+
+function warnIfQueryParametersWereDeleted(
+  incoming: ParsedUrlQuery,
+  rewritten: ParsedUrlQuery
+): void {
+  const incomingQuery = urlQueryToSearchParams(incoming)
+  const rewrittenQuery = urlQueryToSearchParams(rewritten)
+
+  const missingKeys = [...incomingQuery.keys()].filter((key) => {
+    return !rewrittenQuery.has(key)
+  })
+
+  if (missingKeys.length > 0) {
+    Log.warn(
+      `Middleware rewrite deleted the following query parameters from the URL: ${missingKeys.join(
+        ', '
+      )}.`,
+      'We are no longer merging query parameters - https://nextjs.org/docs/messages/errors/deleting-query-params-in-middlewares'
+    )
```

Comment:
```suggestion
    Log.warn(
      `Query params are no longer automatically merged for rewrites in middleware, see more info here: https://nextjs.org/docs/messages/errors/deleting-query-params-in-middlewares'
    )
```
It might be good to wrap this in an `execOnce` so we don't log the warning every time. I think we can also reduce the noise on the warning a bit as the linked error document can do further explaining.